src/actionCreators/index.js

```diff
@@ -41,18 +53,46 @@ export const authorize = (ssoGroup, email) => {
         },
       })
       .then(res => {
+        const userStatus = checkStatus(
+          res.data.isPending,
+          res.data.isAuthorized,
+          res.data.isRejected
+        );
         const newUserInfo = {
+          id: res.data.id,
           email: res.data.email,
           firstName: res.data.firstName,
           lastName: res.data.lastName,
-          authorized: res.data.authorized,
         };
-        if (newUserInfo.authorized) return dispatch(authorizationSuccess(ssoGroup, newUserInfo));
-        return dispatch(authorizationFailed(ssoGroup, newUserInfo));
+        if (userStatus === 2) return dispatch(authorizationSuccess(ssoGroup, newUserInfo));
+        if (userStatus === 3) return dispatch(authorizationFailed(ssoGroup, newUserInfo));
+        return dispatch(authorizationPending(ssoGroup, newUserInfo));
       })
       .catch(err => {
-        console.log(err);
         return dispatch(authorizationError([err.message]));
       });
   };
 };
+
+export const updateUser = (userId, userProfile) => {
+  return dispatch => {
+    dispatch(updateUserStart());
+    axi
+      .put(
+        API.UPDATE_SSO_USER(userId),
+        { ...userProfile },
+        {
+          headers: {
```

Comment:
do the request headers ever change? Maybe we can make a function that creates it, a constant for it, and or have it preconfiged within the axios instance